### ISSUE TITLE
Add smart extensions to the dynamic_route_handler 

### ISSUE THREAD

**Issue by choptastic** [nitrogen/nitrogen_core#60]
**Short:** Allow page modules to support standard (And customizable) extensions. For example: if `http://mysite/my/page/module` was a valid address, then this would allow intelligently retrieving the pertinent data formatted for json by requesting `http://mysite/my/page/module.json` 

**Long:** A `smart_extensions` option should be added to the dynamic_route_handler, which will read a `smart_extensions` value from the config. Smart exceptions in the config should look like this:

``` erlang
{smart_extensions, [
    {"json", fun dynamic_route_handler:json/0},
    {"atom", fun dynamic_route_handler:atom/0},
    {"something_else", fun some_module:some_function/0}
]}.
```

If the extension of a request matches a smart extension:
1. Strip off the extension and see if there is a module that matches the request without an extension. If there is, move on to the next step. If not, this is a static file request and handle it accordingly
2. Re-record the path info sans extension into the context's page_info: `wf_context:page_info(PathinfoSansExt)`
3. Call the function

The function should do something useful with the request, relevant to the extension.  In the default provided ones like (`dynamic_route_handler:json/0`), it would look like this:

``` erlang
dynamic_route_handler:json() ->
   Mod = wf_context:page_module(),
   case erlang:function_exported(Mod, json_body, 0) of
      false ->
         <<"This page does not have a json handler">>;
      true ->
        wf:content_type("application/json"),
        Json = Mod:json_body(),
        _EncodedJson = nitro_mochijson2:encode({struct,Json})
   end.
```

Basically, it would check for the existence of PageModule:json_body function, and if it exists, call it, assuming it returns a proplist-type structure that can be interpreted by json (that way, the programmer doesn't need to deal with json encoding, just returning encodable erlang data), and finally encoding that structure to send it to the client.
